test/python/na/test_state_preparation.py

```diff
@@ -0,0 +1,57 @@
+"""Test the state preparation for zoned neutral atom architectures."""
+
+from __future__ import annotations
+
+from pathlib import Path
+
+import pytest
+from qiskit import QuantumCircuit
+
+from mqt.qmap.na import NAStatePreparationSolver, generate_code, get_ops_for_solver
+
+# get root directory of the project
+circ_dir = Path(__file__).resolve().parent.parent.parent / "na/nasp/circuits"
+
+
+@pytest.fixture
+def solver() -> NAStatePreparationSolver:
+    """Return a NAStatePreparationSolver instance with both sided storage zone."""
+    return NAStatePreparationSolver(3, 7, 2, 3, 2, 2, 2, 2, 2, 4)
+
+
+@pytest.mark.parametrize(
+    "circuit_filename",
+    [
+        "steane.qasm",
+        "surface_3.qasm",
+    ],
+)
+def test_na_state_prep_sat(solver: NAStatePreparationSolver, circuit_filename: str) -> None:
+    """Test the state preparation for the zoned neutral atom architecture."""
+    qc = QuantumCircuit.from_qasm_file(circ_dir / circuit_filename)
+    ops = get_ops_for_solver(qc, "z", 1)
```

Comment:
Similar to the other comment: Should a user really have to call this?
Can't the solver handle this transparently when given a circuit as an input?

Reply:
In contrast to the other comment, I think here, this interface makes the capabilities of the solver clear. Especially, that it can not handle arbitrary circuits and that the entangling gates are taken as a list of qubit pairs. I would be in favour of leaving the interface as is.

Reply:
I can see that point.
It still feels a bit cumbersome to use to be honest.
But I am not willing to die on that hill, so let's just keep it like it is for now 😌